include/edge.h

```diff
@@ -0,0 +1,31 @@
+//
+// Created by Renjie Wu on 2019-02-24.
+//
+
+#ifndef PIPELINE_GRAPH_EDGE_H
+#define PIPELINE_GRAPH_EDGE_H
+
+#include <utility>
+#include <memory>
+#include "node.h"
+namespace pipe_graph {
+    class Edge {
+        typedef std::pair<std::shared_ptr<Node>, std::shared_ptr<Node>> pipe_type;
+        int edge_id_;
+        pipe_type end_points_;
+    public:
+        Edge(int id, std::shared_ptr<Node> n_1, std::shared_ptr<Node> n_2):
+        edge_id_(id), end_points_(n_1,n_2) {}
+
+        //get/set functions
+        double edge_id() const ;
+        void edge_id(int id);
+
+        pipe_type end_points() const;
```

Comment:
Don't return this, as it gives up control of ownership of nodes, breaks encapsulation.